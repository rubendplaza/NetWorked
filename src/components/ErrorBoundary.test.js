import React from "react";
import ErrorBoundary from "./ErrorBoundary";
import { shallow, mount } from "enzyme";

let wrapper;
beforeEach(() => {
  const SomeComponent = () => null;
  wrapper = mount(
    <ErrorBoundary>
      <SomeComponent id="foo" />
    </ErrorBoundary>
  );
});

describe("ErrorBoundary Component", () => {
  test("expect to render ErrorBoundary component", () => {
    expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
  });

  test("testing componentDidCatch should display error message if wrapped component throws", () => {
    const error = new Error("test");
    wrapper.find("#foo").simulateError(error);
    expect(wrapper.state()).toEqual({ hasError: true });
  });

  test("testing componentDidCatch should NOT display error message if wrapped component does not throw", () => {
    expect(wrapper.state()).toEqual({ hasError: false });
  });
});
