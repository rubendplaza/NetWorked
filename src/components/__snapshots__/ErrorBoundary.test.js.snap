// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component expect to render ErrorBoundary component 1`] = `""`;
